lib/tasks/sidekiq.rake

```diff
@@ -21,5 +21,15 @@ namespace :sidekiq do
     end
 
   end
+
+  desc 'Clear sidekiq processed and failed queues to save memory'
+  task :clearQueues => :environment do
+
+    require 'sidekiq/api'
+
+    Sidekiq::Stats.new.reset
+    Sidekiq::DeadSet.new.clear
```

Comment:
Looking at the docs, by default failed jobs are stored in the dead set for 6 months, and a maximum of 10,000 of them. I can understand that if this filled up it could cause an OOM issue, although looking at prod there are only 34 jobs in there at the moment.

I think those numbers can be changed in config too, although I'm not exactly sure how. Maybe a simpler way to handle this if we think that's the issue?

Reply:
OK thanks, will see if I can find a way to do this in the config. We may need to have a discussion around this https://docs.google.com/document/d/1lKfSbCZehTQALoHCvdGZcNwtK65_a_Az-c54-T20YHw/edit# when Ian is back since 34 jobs doesn't seem a lot, yet there were lots on development before I cleared it like thousands in the processed.

Reply:
There is a screenshot of the stats count, but most values are 0, so I presume that was run after the flushdb command was run.
Looking at the memory usage graph in that link, it shows fairly consistent rises for a year. If dead jobs are removed after 6 months I think it would look different.